### ISSUE TITLE
Filter Support for Citext Datatypes

### ISSUE THREAD

**Issue by dbenton9** [activeadmin/activeadmin#6378]
TL;DR: Add `:citext` as supported field detection for filters

I was recently trying to determine why my Active Admin panel was not showing the filter I had set for a field on my database:
```
filter :keyword
```
We discovered a workaround by changing the previous line to:
```
filter :keyword, as: :string
```
It was frustrating that before we fixed the issue Active Admin just appeared to be ignoring the requested filter. I began to dig around the Active Admin code base to discover this https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/filters/forms.rb.

In the `forms.rb` file, default filter types are determined for a variety of different data types, but not for citext. Our field `keyword` is a citext. Thus why it wasn't supported by default. 

Rails began [supporting citext](https://github.com/rails/rails/pull/12523) about 5 years ago.

I'd love to see Active Admin support citexts with a default filter type of `:string` like the datatypes `string` and `text` currently receive. I'm new to contributing, but I read I should get the feature request approved first before I start coding. So this is my feature request.

**Comment by javierjulio** [comment]
Sure, we would be open to a PR that adds `citext` support for filters.
